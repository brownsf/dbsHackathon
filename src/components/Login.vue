<template>
  <v-layout
    row
    wrap>
    <v-flex
      xs12
      md4
      justify-center>
      <v-form
        ref="form"
        v-model="valid"
        lazy-validation>
        <v-text-field
          v-model="email"
          :rules="emailRules"
          label="E-mail"
          required
        />


        <v-text-field
          v-model="password"
          :append-icon="e1 ? 'visibility' : 'visibility_off'"
          :append-icon-cb="() => (e1 = !e1)"
          :type="e1 ? 'text': 'password'"
          :rules="passwordRules"
          name="input-10-1"
          label="Enter your password"
          hint="At least 8 characters"
          min="8"
          counter
          required
        />
        <v-alert
          :value="error"
          type="error">
          {{ errorMessage }}
        </v-alert>

        <v-btn
          :disabled="!valid"
          class="sbButton"
          @click="submit"
        >submit</v-btn>
        <v-btn @click="clear">clear</v-btn>
        <p class="newAccount"><router-link to="register">Create new account</router-link></p>
      </v-form>
    </v-flex>
  </v-layout>
</template>

<style scoped>
.newAccount{
  margin-top: 15px
}
.newAccount a{
  text-decoration: none;
}
</style>


<script>
import { mapState } from 'vuex';

export default {
  data: () => ({
    e1: false,
    valid: true,
    email: '',
    emailRules: [
      v => !!v || 'E-mail is required',
    ],
    password: '',
    passwordRules: [
      v => !!v || 'Password is required',
    ],
  }),
  computed: mapState({
    error: state => state.loginError || false,
    errorMessage: state => state.loginErrorMessage,
  }),
  created() {
    this.$store.dispatch('loginReset');
  },
  methods: {
    submit() {
      if (this.$refs.form.validate()) {
        // Native form submission is not yet supported
        this.$store.dispatch('login', {
          email: this.email,
          password: this.password,
        })
          .then(() => {
            this.$router.push('/');
          });
      }
    },
    clear() {
      this.$refs.form.reset();
    },
  },
};
</script>
