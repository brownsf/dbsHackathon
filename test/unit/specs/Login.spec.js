import { mount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vue from 'vue';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import Login from '@/components/Login';

Vue.use(Vuex);
describe('Register', () => {
  let wrp;

  const routes = [{ path: '/items/:item_id/edit', name: 'item-edit' }];
  const state = { data: {} };
  const actions = {
    register: jest.fn(),
  };

  const store = new Vuex.Store({
    state,
    actions,
  });
  const router = new VueRouter({ routes });

  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(VueRouter);
    localVue.use(Vuetify);

    wrp = mount(Login, {
      localVue,
      router,
    });
  });

  it('should render', () => {
    expect(wrp.html()).toMatchSnapshot();
  });
  it('should have a text input', () => {
    const Constructor = Vue.extend(Login);
    const vm = new Constructor().$mount();
    expect(vm.$el.querySelector('.sbButton').textContent).toEqual('submit');
  });

  it('clicks the button', () => {
    const wrapper = mount(Login, { store });
    wrapper.vm.$refs.form.validate = () => true;
    wrapper.find('.sbButton').trigger('click');
    expect(actions.register.mock.calls).toHaveLength(1);
  });
});
