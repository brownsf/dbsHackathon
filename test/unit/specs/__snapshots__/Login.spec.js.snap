// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render 1`] = `
<div class="layout row wrap">
  <div class="flex xs12 md4 justify-center">
    <form novalidate="novalidate">
      <div class="input-group input-group--required input-group--text-field primary--text">
        <label>E-mail</label>
        <div class="input-group__input">
          <input required="required" tabindex="0" aria-label="E-mail" type="text">
        </div>
        <div class="input-group__details">
          <!---->
        </div>
      </div>
      <div class="input-group input-group--append-icon input-group--required input-group--text-field primary--text">
        <label>Enter your password</label>
        <div class="input-group__input">
          <input name="input-10-1" min="8" required="required" tabindex="0" aria-label="Enter your password" type="text">
          <i aria-hidden="true" class="icon material-icons input-group__append-icon input-group__icon-cb">visibility_off</i>
        </div>
        <div class="input-group__details">
          <!---->
          <div class="input-group__counter">0 / 25</div>
        </div>
      </div>
      <div class="alert error" style="display: none;">
        <i aria-hidden="true" class="icon material-icons alert__icon">warning</i>
        <div>

        </div>
      </div>
      <button type="button" class="sbButton btn">
        <div class="btn__content">submit</div>
      </button>
      <button type="button" class="btn">
        <div class="btn__content">clear</div>
      </button>
      <p class="newAccount">
        <a href="#/register" class="">Create new account</a>
      </p>
    </form>
  </div>
</div>
`;
